feat(voice): use StableLib as a non-native fallback

Snazzah opened the pull request:
Use [StableLib](https://github.com/StableLib/stablelib)'s `@stablelib/xchacha20poly1305` module for non-native support. The package is locked to version 1 due to ESM.

Snazzah committed:
feat(voice): use StableLib as a non-native fallback

TTtie commented on package.json:
```diff
@@ -78,6 +79,9 @@
     "@discordjs/opus": {
       "optional": true
     },
+    "@stablelib/xchacha20poly1305": {
+      "optional": true
+    },
```
nit: `@stablelib/xchacha20poly1305` isn't a peer dependency – this shouldn't be here (not decided on whether to force users to pick between stablelib and https://github.com/projectdysnomia/libsodium yet – that variant is still in play)

Snazzah committed:
fix: remove wrong peer dep

TTtie committed:
nit: don't try to load sodium-native again once stablelib is loaded

TTtie committed:
perf: avoid buffer copying

Buffer.from() copies Uint8Arrays:
```js;
const buf = new Uint8Array(4);
buf[1] = 2;
const buf2 = Buffer.from(buf);
buf2[2] = 1;

console.log(buf[2], buf2[2]);
```
